### ISSUE TITLE
Geth L2 doesn't return revert reason string for external view functions

### ISSUE THREAD

**Issue by krebernisak** [ethereum-optimism/go-ethereum#251]
This worked before but seems to break with the latest update.

Let's define a test function like:

```solidity
function testRequire()
    external
    view
    returns (uint256)
{
    require(false, "This is an revert string");
}
```

If I try to call that function, I'll get an error:

```
Error: call revert exception (method="testRequire()", errorSignature=null, errorArgs=[null], reason=null, code=CALL_EXCEPTION, version=abi/5.0.10)
    at Logger.makeError (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/@ethersproject/logger/lib/index.js:179:21)
    at Logger.throwError (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/@ethersproject/logger/lib/index.js:188:20)
    at Interface.decodeFunctionResult (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/@ethersproject/abi/lib/interface.js:286:23)
    at Contract.<anonymous> (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/ethers/node_modules/@ethersproject/contracts/lib/index.js:319:56)
    at step (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/ethers/node_modules/@ethersproject/contracts/lib/index.js:46:23)
    at Object.next (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/ethers/node_modules/@ethersproject/contracts/lib/index.js:27:53)
    at fulfilled (/Users/krebernisak/Documents/workspace/work/chainlink/code/ovm/optimism-tutorial/node_modules/ethers/node_modules/@ethersproject/contracts/lib/index.js:18:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  reason: null,
  code: 'CALL_EXCEPTION',
  method: 'testRequire()',
  errorSignature: null,
  errorArgs: [ null ],
  address: '0x6dAe38FDddF83f30dD14584002c65cF6a04489Be',
  args: [],
  transaction: {
    data: '0x357815c4',
    to: '0x6dAe38FDddF83f30dD14584002c65cF6a04489Be',
    from: '0x640e7cc27b750144ED08bA09515F3416A988B6a3'
  }
}
```

I'd expect the reason string to be returned in `errorArgs`, but currently `null` is returned both for `reason` and `errorArgs`.

On EVM a message like "VM Exception while processing transaction: revert This is an revert string" is returned.

I've made [a simple project](https://github.com/krebernisak/optimism-tutorial) to demonstrate this behavior.

**Comment by K-Ho** [comment]
@ben-chain pointed out that this resulted from a recent change in how we do `eth_calls`. Will represent this as a ticket on our end to fix

**Comment by krebernisak** [comment]
I can confirm that this is fixed on the latest `364a081` [optimism-integration](https://github.com/ethereum-optimism/optimism-integration). Thanks! Feel free to close the issue.
